src/components/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState } from "react";
+import { generateId } from "../../utils";
+
+interface Props {
```

Comment:
I saw Brian have one of these patterns, I wonder if it is any benefit to us to keep this pattern

```
type IntrinsicElements = JSX.IntrinsicElements["input"];
interface Props extends IntrinsicElements {
```

Reply:
I had that and then removed it. Mainly because I wasn't exactly sure what we were getting from it. I can put it back in.

Reply:
Lets chat with Brian see if he can offer any benefits, if not lets pop it off.